tools/functional-tester/etcd-tester/lease_stresser.go

```diff
@@ -0,0 +1,278 @@
+// Copyright 2015 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/etcdserver/api/v3rpc/rpctypes"
+	pb "github.com/coreos/etcd/etcdserver/etcdserverpb"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+// time to live for lease
+const TTL = 15
+
+type LeaseStresser interface {
+	// Stress starts to stress lease on the etcd cluster
+	Stress() error
+	// Cancel cancels the lease stress test on the etcd cluster
+	Cancel()
+	// Report reports the success and failure of the lease stress test
+	Report() (success int, failure int)
+	// Setup initializes fields of leaseStresser struct
+	Setup() error
+}
+
+type leaseStressConfig struct {
+	numOfLeases  int
+	keysPerLease int
+}
+
+type leaseStresser struct {
+	endpoint string
+	cancel   func()
+	conn     *grpc.ClientConn
+	kvc      pb.KVClient
+	lc       pb.LeaseClient
+
+	success      int
+	failure      int
+	numOfLeases  int
+	keysPerLease int
+
+	// mu Mutex is used to protect the shared maps aliveLeases and revokedLeases
+	// which are accessed and modified by different go routines.
+	mu            sync.Mutex
+	aliveLeases   map[int64]struct{}
+	revokedLeases map[int64]struct{}
+
+	runWg *sync.WaitGroup
```

Comment:
I follow the way that stresser define the wait group. https://github.com/fanminshi/etcd/blob/lease_stressor/tools/functional-tester/etcd-tester/stresser.go#L145.
I wonder why using pointer in my case is not a good idea.